### PR TITLE
Function support at the binder level

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Allow Outbound channel to be configured with the function support at the MessageChannel binder level
 TBD: Configure inbound channel
 - Refactor the IntegrationFlowFunctionSupport methods
 - Update tests

### CODE REVIEW

**Comment by artembilan** [comment]
According Travis report, there is a test failure:
```
testLateBindingConsumer(org.springframework.cloud.stream.binding.BindingServiceTests)  Time elapsed: 121.08 sec  <<< FAILURE!
java.lang.AssertionError: 
Expecting:
 <Mock for Binding, hashCode: 1353052385>
and actual:
 <null>
to refer to the same object
	at org.springframework.cloud.stream.binding.BindingServiceTests.testLateBindingConsumer(BindingServiceTests.java:480)
```

Looks like sporadic, but as experience shows it always a result of some race condition...

**Comment by ilayaperumalg** [comment]
yeah, My local tests ran fine as well.

File: spring-cloud-stream/src/test/java/org/springframework/cloud/stream/function/SourceToFunctionsSupportTests.java [hunk 1]
```diff
@@ -70,7 +71,7 @@
 	@Test
 	public void testFunctionIsAppliedToExistingMessageSource() {
 		try (ConfigurableApplicationContext context = new SpringApplicationBuilder(
-			TestChannelBinderConfiguration.getCompleteConfiguration(FunctionsConfiguration.class)).web(
+			TestChannelBinderConfiguration.getCompleteConfiguration(FunctionsConfiguration.class, TestChannelBinderConfiguration.class)).web(
```
Comment (olegz) [inline spring-cloud-stream/src/test/java/org/springframework/cloud/stream/function/SourceToFunctionsSupportTests.java]: Why do you need to add `TestChannelBinderConfiguration` to `TestChannelBinderConfiguration`?

Comment (ilayaperumalg) [reply spring-cloud-stream/src/test/java/org/springframework/cloud/stream/function/SourceToFunctionsSupportTests.java]: This was an oversight.

**Comment by dturanski** [comment]
This PR causes a Spring lifecycle issue when testing with Sftp source, a fairly complex scenario, but the fix requires a different configuration approach. I am closing this PR and replacing with a new one that works around this problem by configuring `IntegrationFlowFunctionSupport` in `AbstractBinder.afterPropertiesSet`

**Pull request closed without merging**